File: common/src/main/java/generations/gg/generations/core/generationscore/world/level/block/EnchantedObsidianBlock.java
package generations.gg.generations.core.generationscore.world.level.block;

import net.minecraft.core.BlockPos;
import net.minecraft.world.level.BlockGetter;
import net.minecraft.world.level.block.Block;
import net.minecraft.world.level.block.state.BlockState;

/**
 * @author JT122406
 * @see net.minecraftforge.common.extensions.IForgeBlock#isPortalFrame(BlockState, BlockGetter, BlockPos)
 * Allows EnchantedObsidian to be used as a nether portal frame
 */
public class EnchantedObsidianBlock extends Block implements IPortalBlock {

    public EnchantedObsidianBlock(Properties arg) {
        super(arg);
    }

    public boolean isPortalFrame(BlockState state, BlockGetter level, BlockPos pos) {
        return state.is(GenerationsBlocks.ENCHANTED_OBSIDIAN_SET.getBaseBlock());
    }
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/EnchantedObsidianBlock.kt
package generations.gg.generations.core.generationscore.common.world.level.block

import net.minecraft.core.BlockPos
import net.minecraft.world.level.BlockGetter
import net.minecraft.world.level.block.Block
import net.minecraft.world.level.block.state.BlockState

/**
 * @author JT122406
 * @see net.minecraftforge.common.extensions.IForgeBlock.isPortalFrame
 */
class EnchantedObsidianBlock(arg: Properties) : Block(arg),
    IPortalBlock {
    override fun isPortalFrame(state: BlockState, level: BlockGetter, pos: BlockPos): Boolean {
        return state.`is`(GenerationsBlocks.ENCHANTED_OBSIDIAN_SET.getBaseBlock())
    }
}

